RxCocoa/Common/DelegateProxyType.swift

```diff
@@ -86,56 +86,98 @@ and extend it
 
 */
 public protocol DelegateProxyType : AnyObject {
-    /// DelegateProxy factory
+    associatedtype ParentObject: AnyObject
+    associatedtype Delegate: AnyObject
+    
+    /// DelegateProxy factory. Use 'DelegateProxyFactory.sharedFactory'.
     static var factory: DelegateProxyFactory { get }
 
+    /// Creates new proxy for target object.
+    /// Should not call this function directory, use 'DelegateProxy.proxyForObject'
+    static func createProxy(for object: AnyObject) -> AnyObject
```

Comment:
I think we can remove this from `DelgateProxyType` since protocol extension is already implementing it depending on `factory`.

Reply:
Shouldn't this now have the following signature?
`static func createProxy(for object: ParentObject) -> Self`

Reply:
If I use `static func createProxy(for object: ParentObject) -> Self` instead, 
I can see compile error like `Method 'createProxy(for:)' in non-final class 'RxScrollViewDelegate'`.